src/Microsoft.AspNet.TestHost/ClientHandler.cs

```diff
@@ -218,6 +218,7 @@ internal void Abort(Exception exception)
 
             public void Dispose()
             {
+                _responseFeature.FireOnResponseCompleted();
```

Comment:
This is not the end of the response, there may be multiple requests/responses per ClientHandler.